base/files/file_starboard.cc

```diff
@@ -270,44 +277,67 @@ bool File::GetInfo(Info* info) {
 }
 
 File::Error File::GetLastFileError() {
-  int err = errno;
-  switch (err) {
-    case ENOENT:
-      return static_cast<File::Error>(kSbFileErrorNotFound);
-    case EACCES:
-      return static_cast<File::Error>(kSbFileErrorAccessDenied);
-    case EEXIST:
-      return static_cast<File::Error>(kSbFileErrorExists);
-    default:
-      NOTREACHED() << base::StringPrintf("Unhandled errno: %d", err);
-  }
-  return FILE_ERROR_FAILED;
+  return base::File::OSErrorToFileError(g_sb_file_error);
 }
 
 // Static.
-File::Error File::OSErrorToFileError(SbSystemError sb_error) {
-  switch (sb_error) {
+File::Error File::OSErrorToFileError(SbSystemError sb_system_error) {
+  switch (static_cast<SbFileError>(sb_system_error)) {
+    case kSbFileOk:
+      static_assert(static_cast<int>(kSbFileOk) == static_cast<int>(FILE_OK));
```

Comment:
Do the static_assert statements need to be part of the switch statement?

Reply:
Of course doesn't have to be, but this way makes it obvious if ever new cases are added that it should be checked.

Reply:
Honestly, it may be safer and cleaner to not assume the mapping is 1:1 and just return the FILE_FOO value everywhere...